x-pack/plugins/ml/public/application/jobs/components/custom_url_editor/utils.js

```diff
@@ -152,10 +150,10 @@ function buildDashboardUrlFromSettings(settings) {
           query = queryFromEntityFieldNames;
         }
 
-        const getUrlGenerator = getGetUrlGenerator();
-        const generator = getUrlGenerator(DASHBOARD_APP_URL_GENERATOR);
-        return generator
-          .createUrl({
+        const dashboardLocator = getDashboardLocator();
+        if (!dashboardLocator) return reject();
```

Comment:
super duper nit: Would be good to throw an error or `console.error` here indicating that `dashboardLocator` is not available before rejecting so it's easier to troubleshoot if something fails.

Reply:
Good call! Updated that!